packages/ws-client/src/data-source.ts

```diff
@@ -162,17 +156,15 @@ export function createWSDataSource(wsUrl: string): WebSocketDataSource {
     state: createStateStream(rws),
     sendEvent: async (event) => {
       send({ $: 'evt', event })
-      if (isHandlerEvent(event)) {
-        return new Promise((resolve, reject) => {
-          promises.set(event.dataState.key, [resolve, reject])
-          setTimeout(() => {
-            if (promises.has(event.dataState.key)) {
-              reject(new Error('Request timeout'))
-              promises.delete(event.dataState.key)
-            }
-          }, event.dataState.timeout || 10_000)
-        })
-      }
+      return new Promise((resolve, reject) => {
+        promises.set(event.dataState.key, resolve)
+        setTimeout(() => {
```

Comment:
do we need to clear this timeout?

Reply:
I don't think so - but I am not an expert. There's a check if event has been already resolved inside the timeout, so it will get called for each event and just noop. Maybe it could be a good practice to set the timeout on the map and clear it.

Reply:
Decided on Discord it's not a big deal for now.